api/errors/error.go

```diff
@@ -0,0 +1,25 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker engine but are not tied to any specific top-level component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+const errGroup = "engine"
+
+var (
+	// ErrorCodeGeneric is generated when there is no better error message
+	// available. This will typically happen when some low-level function
+	// is called and we just need to bubble the unknown error back up to
+	// the API. We should try to avoid using this.
+	ErrorCodeGeneric = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
The error message from the generic errorcode isn't really that helpful in this case so I made one that's better suited for the engine.  To talk about "API classifications" won't really mean much to the end-user and, in fact, could make it sound like there's an error in Docker because it didn't "classify" all of its possible errors.  Granted, its subjective. A generic package like errorcode can't assume the text it picked for its canned errors are good enough for everyone in all cases, and in this case its not.  I can rename it if you want so it doesn't look like we're duplicating the errocode one, just having a hard time thinking of a choice.